homeassistant/components/device_tracker/automatic.py

```diff
@@ -52,64 +60,172 @@ def async_setup_scanner(hass, config, async_see, discovery_info=None):
+                self.ws_reconnect_handle = async_track_time_interval(
+                    self.hass, self.ws_connect, timedelta(minutes=5))
+            return
+
+        if self.ws_reconnect_handle is not None:
+            self.ws_reconnect_handle()
+            self.ws_reconnect_handle = None
 
-        _LOGGER.debug('Updating devices %s', now)
+        _LOGGER.info("Websocket connected.")
 
         try:
-            vehicles = yield from self.session.get_vehicles()
+            yield from ws_loop_future
         except aioautomatic.exceptions.AutomaticError as err:
             _LOGGER.error(str(err))
-            return False
 
-        for vehicle in vehicles:
-            name = vehicle.display_name
-            if name is None:
-                name = ' '.join(filter(None, (
-                    str(vehicle.year), vehicle.make, vehicle.model)))
+        _LOGGER.info("Websocket closed.")
 
-            if self.devices is not None and name not in self.devices:
-                continue
+        # If websocket was close was not requested, attempt to reconnect
+        if not self.ws_close_requested:
+            self.hass.loop.create_task(self.ws_connect())
 
-            self.hass.async_add_job(self.update_vehicle(vehicle, name))
+    @asyncio.coroutine
+    def ws_close(self):
+        """Close the websocket connection."""
+        self.ws_close_requested = True
+        if self.ws_reconnect_handle is not None:
+            self.ws_reconnect_handle()
+            self.ws_reconnect_handle = None
+
+        yield from self.client.ws_close()
+
+    @asyncio.coroutine
+    def load_vehicle(self, vehicle):
+        """Load the vehicle's initial state and update hass."""
+        kwargs = yield from self.get_vehicle_info(vehicle)
+        yield from self.async_see(**kwargs)
 
     @asyncio.coroutine
-    def update_vehicle(self, vehicle, name):
-        """Update the specified vehicle's data."""
+    def get_vehicle_info(self, vehicle):
+        """Fetch the latest vehicle info from automatic."""
         import aioautomatic
 
-        kwargs = {
-            'dev_id': vehicle.id,
-            'host_name': name,
-            'mac': vehicle.id,
-            ATTR_ATTRIBUTES: {
-                'fuel_level': vehicle.fuel_level_percent,
-                }
-        }
+        name = vehicle.display_name
+        if name is None:
+            name = ' '.join(filter(None, (
+                str(vehicle.year), vehicle.make, vehicle.model)))
+
+        if self.devices is not None and name not in self.devices:
+            self.vehicle_info[vehicle.id] = None
+            return
+        else:
+            self.vehicle_info[vehicle.id] = kwargs = {
+                ATTR_DEV_ID: vehicle.id,
+                ATTR_HOST_NAME: name,
+                ATTR_MAC: vehicle.id,
+                ATTR_ATTRIBUTES: {
+                    ATTR_FUEL_LEVEL: vehicle.fuel_level_percent,
+                    }
```

Comment:
The formatting is validated by the linter and its indented to distinguish it from the next closing brace. Can you explain what you'd prefer to see?